jacodb-ets/src/test/kotlin/org/jacodb/ets/test/utils/Entrypoints.kt

```
@@ -0,0 +1,133 @@
+            etsFile.dumpDot(file)
+        }
+    }
+}
+
+@OptIn(ExperimentalPathApi::class)
+object DumpEtsFilesToDot {
+    private const val DIR = "/etsir/samples/"
+    private const val DOT_DIR = "dot"
+
+    @JvmStatic
+    fun main(args: Array<String>) {
+        val baseDirUrl = object {}::class.java.getResource(DIR)
+            ?: error("Resource not found: '$DIR'")
+        val baseDir = Paths.get(baseDirUrl.toURI())
+
+        val dotDir = baseDir.resolveSibling(DOT_DIR)
+        if (!dotDir.exists()) {
+            dotDir.createDirectories()
+        }
+
+        baseDir.walk()
+            .filter { it.name.endsWith(".json") }
+            .forEach { path ->
+                val relativePath = baseDir.relativize(path)
+                val fileName = path.nameWithoutExtension
+
+                process(relativePath, fileName, dotDir, "Dto") {
+                    loadEtsFileDtoFromResource(it)
+                }
+                process(relativePath, fileName, dotDir, "") {
+                    loadEtsFileFromResource(it)
+                }
+
+                logger.info { "Processed: $path" }
+            }
+    }
+
+    private fun <T> process(
+        path: Path,
+        name: String,
+        dotDir: Path,
+        suffix: String,
+        load: (String) -> T,
+    ) {
+        try {
+            val resourcePath = DIR + path.toString()
+            val fileData = load(resourcePath)
+            val outputDir = dotDir.resolve(path.parent ?: dotDir)
+            if (!outputDir.exists()) {
+                outputDir.createDirectories()
+            }
+            val outputPath = outputDir.resolve("${name}${suffix}.dot").toString()
+            render(outputPath) { file ->
+                when (fileData) {
+                    is EtsFileDto -> fileData.dumpDot(file)
+                    is EtsFile -> fileData.dumpDot(file)
+                }
+            }
+        } catch (e: Exception) {
+            logger.error { e.stackTraceToString() }
```

These entrypoints are to be called manually anyway. Why not just let them unwind on exception (i.e. remove try/catch here) instead of semi-silently ignoring them? Imagine the situation: you enter all the necessary paths to the files you need to be rendered, hit Run, await, receive green checkmark in IDEA, go open the resulting PDFs. Most probably, you just don't notice than an exception has occurred.